### PR TITLE
Python: Fix broken concept samples. Run concept examples as tests.

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

We had many broken concept examples. These samples are also not covered by tests so it's hard to know when things are out of sync.

### Description

Fix the Python concept samples, and also cover them with tests so we aren't checking in code that is going to break them.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py">open_ai_chat_completion_base.py</a></td><td>219</td><td>96</td><td>56%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L97">97</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L101">101</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L121">121</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L146-L150">146&ndash;150</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L174">174</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L178">178</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L182">182</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L201-L206">201&ndash;206</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L223-L251">223&ndash;251</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L254-L265">254&ndash;265</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L280-L287">280&ndash;287</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L298-L306">298&ndash;306</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L322-L329">322&ndash;329</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L350">350</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L358">358</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L364-L367">364&ndash;367</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L379-L382">379&ndash;382</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L413">413</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L452">452</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L457">457</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L462-L467">462&ndash;467</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L471-L478">471&ndash;478</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L482-L494">482&ndash;494</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L516">516</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L526-L537">526&ndash;537</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/search_engine</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/search_engine/bing_connector.py">bing_connector.py</a></td><td>33</td><td>22</td><td>33%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/search_engine/bing_connector.py#L38">38</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/c4d582e7e774979c78e419c22f5309d7d93e4952/python/semantic_kernel/connectors/search_engine/bing_connector.py#L47-L91">47&ndash;91</a></td></tr><tr><td><b>TOTAL</b></td><td><b>6140</b></td><td><b>717</b></td><td><b>88%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1434 | 1 :zzz: | 0 :x: | 0 :fire: | 23.592s :stopwatch: |

File: python/samples/concepts/prompt_templates/load_yaml_prompt.py [hunk 1]
```diff
@@ -13,15 +13,18 @@ async def main():
 
     service_id = "default"
     chat_service = OpenAIChatCompletion(
-        ai_model_id="gpt-4-0613",
+        ai_model_id="gpt-3.5-turbo-1106",
```
Comment (glahaye) [inline python/samples/concepts/prompt_templates/load_yaml_prompt.py:16]: We want this older model instead of the newer one?

**Review by glahaye — APPROVED** [review]

File: python/samples/concepts/prompt_templates/load_yaml_prompt.py (hunk 1 above)
Comment (moonbox3) [reply python/samples/concepts/prompt_templates/load_yaml_prompt.py:16]: No need for a powerful model for this, and isn't gpt-3.5 cheaper? :)

File: python/.env.example [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 OPENAI_ORG_ID=""
-AZURE_OPENAI_SYSTEM_MESSAGE="You are an AI assistant that helps people find information"
-AZURE_OPENAI_API_VERSION="2024-02-15-preview"
-AZURE_OPENAI_DEPLOYMENT_NAME=""
+AZURE_OPENAI_CHAT_DEPLOYMENT_NAME=""
+AZURE_OPENAI_TEXT_DEPLOYMENT_NAME=""
+AZURE_OPENAI_EMBEDDING_DEPLOYMENT_NAME=""
 AZURE_OPENAI_ENDPOINT=""
 AZURE_OPENAI_API_KEY=""
+AZURE_OPENAI_SYSTEM_MESSAGE="You are an AI assistant that helps people find information"
```
Comment (eavanvalkenburg) [inline python/.env.example:11]: do we actually read this one from our code? if not let's either implement that, or delete from this example!

**Review by eavanvalkenburg — APPROVED** [review]
small note, otherwise looks good!

**Review by eavanvalkenburg — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**